app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,80 @@
+class OrdersController < ApplicationController
+  def show
+    id = params[:id]
+    @order = Order.find(id)
+    if @current_user != nil
+      if @current_user.orders.include?(@order)
+        @order_items = @current_user.order_items.where(order_id: id)
+      else
+        redirect_to merchant_path(@current_user)
+      end
+    else
+    redirect_to root_path
+    end
+    # flash[:error] = "Sorry, you are not authorized to view the page you were trying to see.  Here is a better page for you. "
```

Comment:
It would be good to clean up irrelevant/working code comments before you make a PR